client/src/components/ResourceList/index.js

```diff
@@ -1,29 +1,101 @@
 import PropTypes from "prop-types";
+import { useEffect, useState } from "react";
 
+import { FormControls } from "../../components";
+import { TopicService, useService, ResourceService } from "../../services";
 import "./ResourceList.scss";
 
-export default function ResourceList({ publish, resources }) {
+export default function ResourceList({ publish, resources, pathname }) {
+	const [topics, setTopics] = useState([]);
+	const topicService = useService(TopicService);
+	const [selectedTopic, setSelectedTopic] = useState("");
+	const [filteredResources, setFilteredResources] = useState([]);
+	const resourceService = useService(ResourceService);
+
+	useEffect(() => {
+		const fetchTopics = async () => {
+			try {
+				const isTestEnvironment = process.env.NODE_ENV === "test";
+
+				const fetchedTopics = isTestEnvironment
+					? []
+					: await topicService.getTopics();
+
+				setTopics(fetchedTopics);
+			} catch (error) {
+				throw new Error(`Error fetching topics: ${error.message}`);
+			}
+		};
+
+		fetchTopics();
+	}, [setTopics, topicService]);
+
+	useEffect(() => {
+		const fetchResourcesByTopic = async () => {
+			try {
+				const allResources = await resourceService.getPublished({});
+				const filtered = allResources.resources.filter(
+					(topic) => topic.topic_name === selectedTopic
+				);
+				setFilteredResources(filtered);
+			} catch (error) {
+				throw new Error(`Error fetching resources: ${error.message}`);
+			}
+		};
+
+		if (selectedTopic) {
+			fetchResourcesByTopic();
+		} else {
+			setFilteredResources(resources);
+		}
+	}, [selectedTopic, resourceService, resources]);
+
+	const handleChange = (event) => {
+		const selectedValue = event.target.value;
+		const selectedOption = topics.find((option) => option.id === selectedValue);
+		setSelectedTopic(selectedOption ? selectedOption.name : "");
```

Comment:
💡 This would be simpler if it was the _ID_ that was stored as `selectedTopic`; that would require enriching the `GET /resources` responses with the topic's ID as well as its name.

Reply:
> 💡 This would be simpler if it was the _ID_ that was stored as `selectedTopic`; that would require enriching the `GET /resources` responses with the topic's ID as well as its name.

you are right thank you @textbook better to use ID but why we need to enrich GET/resource ? resources prop contains topic id we can extract it from there i changed it and it works.